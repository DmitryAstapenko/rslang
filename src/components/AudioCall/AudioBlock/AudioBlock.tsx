import React, { FunctionComponent, useEffect } from "react";
import VolumeDownIcon from "@material-ui/icons/VolumeDown";
import useSound from "use-sound";
import "./AudioBlock.scss";
import { BACKEND_PATH } from "../../../constants/request-params";
import useKeyDown from "../../../hooks/useKeyDown";

interface IAudioProps {
  audio: string;
  image: string;
  isAnswer: boolean;
  word: string;
}

const AudioBlock: FunctionComponent<IAudioProps> = ({
  audio,
  image = "",
  isAnswer = false,
  word = "",
}: IAudioProps) => {
  const [play] = useSound(`${BACKEND_PATH}${audio}`);
  useEffect(() => {
    play();
  }, [play]);

  useKeyDown("Space", play);

  const imageElement = (
    <img
      src={`${BACKEND_PATH}${image}`}
      className="audio-block-image"
      alt="answer-img"
    />
  );
  if (isAnswer) {
    return (
      <>
        {imageElement}
        <div className="audio-block-container">
          <button
            type="button"
            className="audio-block-answer"
            onClick={() => play()}
          >
            <VolumeDownIcon className="audio-block-icon" />
          </button>
          <span className="audio-block-word">{word}</span>
        </div>
      </>
    );
  }
  return (
    <button type="button" className="audio-block" onClick={() => play()}>
      <VolumeDownIcon className="audio-block-icon" />
    </button>
  );
};

export default AudioBlock;
