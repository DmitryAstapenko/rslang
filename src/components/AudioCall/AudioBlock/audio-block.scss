.audio-block {
  width: 100px;
  height: 100px;
  border-radius: 50%;
  border: 2px solid black;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(61, 7, 189, 0.3);
}
.audio-block:focus, .audio-block-answer:focus{
  outline: none;
}
.audio-block:hover, .audio-block-answer:hover{
  cursor: pointer;
  background-color: rgba(61, 7, 189, 0.7);
}
.audio-block-icon{
  width: 70% !important;
  height: 70% !important;
}
.audio-block-image{
  width: 200px;
  height: 120px;
}
.audio-block-word{
  font-size: 24px;
  font-weight: 700;
}
.audio-block-container{
  margin-top: 10px;
  display: flex;
  justify-content: center;
  align-items: center;
}

.audio-block-answer {
  width: 45px;
  height: 45px;
  border-radius: 50%;
  border: 2px solid black;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(61, 7, 189, 0.3);
  margin-right: 5px;
}

