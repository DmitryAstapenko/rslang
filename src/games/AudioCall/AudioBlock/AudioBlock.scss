.audio-block {   
  margin-bottom: 72px !important;
  width: 120px !important;
  height: 120px !important;  
  animation: show 0.5s forwards;
}

@keyframes show {
  0%   {transform: scale(0); opacity: 0;}  
  100% {transform: scale(1); opacity: 1;}
}

.audio-block-icon{
  width: 70% !important;
  height: 70% !important;
}

.audio-block-image{
  width: 120px !important;
  height: 120px !important;
  animation: show 0.5s forwards;
}

.audio-block-word{  
  font-weight: bold !important;
}

.audio-block-answer {
  margin-right: 1rem !important;
}

.audio-block-container{  
  display: flex;
  justify-content: space-between;
  align-items: center;
  animation: show 0.5s forwards;
}

.audio-block-answer {
  width: 45px;
  height: 45px;
  border-radius: 50%;
  border: 2px solid black;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: rgba(61, 7, 189, 0.3);
  margin-right: 5px;
}

