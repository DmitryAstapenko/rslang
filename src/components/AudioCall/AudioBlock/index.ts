import AudioBlock from "./AudioBlock";

export default AudioBlock;
