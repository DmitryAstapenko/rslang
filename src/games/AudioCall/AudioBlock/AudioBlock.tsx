import React, { FunctionComponent, useEffect } from "react";
import VolumeDownIcon from "@material-ui/icons/VolumeDown";
import useSound from "use-sound";
import "./AudioBlock.scss";
import { Box } from "@material-ui/core";
import { BACKEND_PATH } from "../../../constants/request-params";
import useKeyDown from "../../../hooks/useKeyDown";

interface IAudioProps {
  audio: string;
  image: string;
  isAnswer: boolean;
  word: string;
}

const AudioBlock: FunctionComponent<IAudioProps> = ({
  audio,
  image = "",
  isAnswer = false,
  word = "",
}: IAudioProps) => {
  const [play] = useSound(`${BACKEND_PATH}${audio}`);
  useEffect(() => {
    play();
  }, [play]);

  useKeyDown("Space", play);

  const imageElement = (
    <img
      src={`${BACKEND_PATH}${image}`}
      className="audio-block-image"
      alt="answer-img"
    />
  );

  const SpaceLabel = (
    <Box p={2} mt={1}>
      Press &quot;SPACE&quot; to play the word
    </Box>
  );

  if (isAnswer) {
    return (
      <>
        {imageElement}
        <div className="audio-block-container">
          <button
            type="button"
            className="audio-block-answer"
            onClick={() => play()}
          >
            <VolumeDownIcon className="audio-block-icon" />
          </button>
          <span className="audio-block-word">{word}</span>
        </div>
        {SpaceLabel}
      </>
    );
  }
  return (
    <>
      <button type="button" className="audio-block" onClick={() => play()}>
        <VolumeDownIcon className="audio-block-icon" />
      </button>
      {SpaceLabel}
    </>
  );
};

export default AudioBlock;
