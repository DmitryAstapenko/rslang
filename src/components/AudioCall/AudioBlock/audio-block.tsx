import React, { FunctionComponent, useEffect } from "react";
import VolumeDownIcon from "@material-ui/icons/VolumeDown";
import useSound from "use-sound";
import "./audio-block.scss";

interface IAudioProps {
  audio: string;
  image: string;
  isAnswer: boolean;
  word: string;
}

const AudioBlock: FunctionComponent<IAudioProps> = ({
  audio = "",
  image = "",
  isAnswer = false,
  word = "",
}: IAudioProps) => {
  const backendPath = "https://rnovikov-rs-lang-back.herokuapp.com/";
  const [play] = useSound(`${backendPath}${audio}`);
  useEffect(() => {
    play();
  }, [play]);
  const imageElement = (
    <img
      src={`${backendPath}${image}`}
      className="audio-block-image"
      alt="answer-img"
    />
  );
  if (isAnswer) {
    return (
      <>
        {imageElement}
        <div className="audio-block-container">
          <button
            type="button"
            className="audio-block-answer"
            onClick={() => play()}
          >
            <VolumeDownIcon className="audio-block-icon" />
          </button>
          <span className="audio-block-word">{word}</span>
        </div>
      </>
    );
  }
  return (
    <button type="button" className="audio-block" onClick={() => play()}>
      <VolumeDownIcon className="audio-block-icon" />
    </button>
  );
};

export default AudioBlock;
