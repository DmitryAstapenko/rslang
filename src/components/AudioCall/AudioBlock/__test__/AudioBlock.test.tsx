import React from "react";
import AudioBlock from "../AudioBlock";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

interface IAudioProps {
  audio: string;
  image: string;
  isAnswer: boolean;
  word: string;
}

const setUp = (props:IAudioProps) => shallow(<AudioBlock {...props} />);

describe("should render Score component", () => {
  let component:any;
  beforeEach(() => {
    const audio: string = "path-audio";
    const image: string = "path-image";
    const isAnswer: boolean = false;
    const word: string = "current-word";
    component = setUp({audio, image, isAnswer, word});
  });
  it("should contain audio block icon", () => {
    const wrapper = component.find(".audio-block-icon");
    expect(wrapper.length).toBe(1);
  });
  it("should render AudioBlock component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
