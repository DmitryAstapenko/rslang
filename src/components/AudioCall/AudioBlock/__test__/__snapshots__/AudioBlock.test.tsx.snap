// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Score component should render AudioBlock component 1`] = `
<Fragment>
  <button
    className="audio-block"
    onClick={[Function]}
    type="button"
  >
    <Memo(ForwardRef(VolumeDownIcon))
      className="audio-block-icon"
    />
  </button>
  <Styled(MuiBox)
    mt={1}
    p={2}
  >
    Press "SPACE" to play the word
  </Styled(MuiBox)>
</Fragment>
`;
